paddle/phi/backends/onednn/matmul_utils.h

```diff
@@ -0,0 +1,403 @@
+                   const DenseTensor& x,
+                   const DenseTensor& y,
+                   const std::vector<int64_t>& x_dims,
+                   const std::vector<int64_t>& y_dims,
+                   bool trans_x,
+                   bool trans_y,
+                   DenseTensor* out) {
+  auto shape_x = dev_ctx.HasDnnAttr("fused_reshape_X")
+                     ? PADDLE_GET_CONST(std::vector<int>,
+                                        dev_ctx.GetDnnAttr("fused_reshape_X"))
+                     : std::vector<int>();
+  auto axis_x = dev_ctx.HasDnnAttr("fused_transpose_X")
+                    ? PADDLE_GET_CONST(std::vector<int>,
+                                       dev_ctx.GetDnnAttr("fused_transpose_X"))
+                    : std::vector<int>();
+  auto shape_y = dev_ctx.HasDnnAttr("fused_reshape_Y")
+                     ? PADDLE_GET_CONST(std::vector<int>,
+                                        dev_ctx.GetDnnAttr("fused_reshape_Y"))
+                     : std::vector<int>();
+  auto axis_y = dev_ctx.HasDnnAttr("fused_transpose_Y")
+                    ? PADDLE_GET_CONST(std::vector<int>,
+                                       dev_ctx.GetDnnAttr("fused_transpose_Y"))
+                    : std::vector<int>();
+
+  auto x_strides_override =
+      GetInputStrides("X", x.dims(), trans_x, shape_x, shape_x);
+  auto y_strides_override =
+      GetInputStrides("Y", y.dims(), trans_y, shape_y, axis_y);
+  MatmulOneDNNHandler<T, T, T_out> handler(dev_ctx,
+                                           x_dims,
+                                           y_dims,
+                                           trans_x,
+                                           trans_y,
+                                           x_strides_override,
+                                           y_strides_override,
+                                           IsOutputFused(dev_ctx));
+
+  const auto src_memory_p = handler.AcquireSrcMemory(&x);
+  const auto weights_memory_p = handler.AcquireWeightsMemory(&y);
+  const auto dst_memory_p = handler.AcquireDstMemory(dev_ctx, out);
+
+  auto matmul_p = handler.AcquireForwardPrimitive();
+
+  std::unordered_map<int, memory> matmul_args = {
+      {DNNL_ARG_SRC, *src_memory_p},
+      {DNNL_ARG_WEIGHTS, *weights_memory_p},
+      {DNNL_ARG_DST, *dst_memory_p}};
+
+  const auto* residual_data = dev_ctx.HasDnnInput("ResidualData")
+                                  ? dev_ctx.GetDnnInput("ResidualData")
+                                  : nullptr;
+
+  if (residual_data) {
+    const auto residual_data_memory_p = handler.AcquireSrcMemory(residual_data);
+    matmul_args.insert({DNNL_ARG_ATTR_MULTIPLE_POST_OP(0) | DNNL_ARG_SRC_1,
+                        *residual_data_memory_p});
+  }
+
+  auto& astream = OneDNNContext::tls().get_stream();
+  matmul_p->execute(astream, matmul_args);
+  astream.wait();
+
+  // TODO(jczaja): Explain why int8 format of dst is ABCD and do not need
```

Comment:
@jczaja could you explain this?

Reply:
It is a question that came up in my head during debugging some longer time ago. I do not have more info than what is in the comment

Reply:
The code is not a place for such comments...
Please create a ticket to explain it and remove this TODO from the code.